tko.io/src/3to4.md

```diff
@@ -0,0 +1,249 @@
+| iterable `observableArray`   | Yes  | Browsers with [Symbol support](https://kangax.github.io/compat-table/es6/#test-Symbol)  | No  |
+| event `throttle` / `debounce`   | Yes  | Yes  | No  |
+| `ko.Component`   | Yes | Yes  | No  |
+| `applyBindings` returns `Promise`   | Yes  | Yes  | No  |
+| Polyfills included   | None  | Up to ES6  |  `Object.prototype.bind`  |
+| `ko.proxy`   |  Yes  |  No  |  No  |
+| `ko.onError` | Does not automatically throw (you need to add `throw error` to `ko.options.onError`) | Does not rethrow automatically | Always re-throws error |
+| `foreach` binding | FastForEach | Template foreach | Template foreach |
+| `each` binding    | FastForEach | FastForEach | Template foreach
+
+# Security advantages in TKO
+TKO is ready for strict Content Security Policies (CSP) and does not use `evil-eval` or `new function` techniques. It also provides HTML sanitization through configurable options. To sanitize HTML templates, override the `options.sanitizeHtmlTemplate` setting using a library such as DOMPurify or Validator.js.
+# Build System
+
+TKO has a complete build system rewrite and a conversion of the code from a concatenation of source files to discrete ES6 modules with `import` and `export` statements.  This process has been complex and taken years, with the result being a Knockout with a strong foundation of discrete reusable components.
+
+Knockout 4 and TKO will be maintained at the new monorepo [knockout/tko](https://github.com/knockout/tko) on GitHub.  Knockout 3.x will continue to be maintained at [knockout/knockout](https://github.com/knockout/knockout).
+
+TKO now uses standard tools like `lerna` and `yarn`, and employs [StandardJS](https://standardjs.com/) for code style — though all the old code is not yet updated to StandardJS yet.
+
+# Test System
+
+Knockout 3 used Jasmine 1 to perform tests.  Some packages in TKO continue to use Jasmine 1, while others now use Mocha.
+
+## Backwards compatibility and 💥 breaking changes
+
+We've strived to have backwards compatibility, and the numerous test cases we've run it through have been successful.  A key value of Knockout has been and remains the comprehensive test cases that enforce performance to a predictable standard, and TKO meets all the tests that were enforced on Knockout 3.x, notwithstanding a few issues:
+
+- The `function` no longer works in `data-bind`, having been replaced by lambdas, so `data-bind='click: function () { x++ }'` will have to be replaced by e.g. `data-bind='click: => x++'`
+- The `data-bind` is now interpreted by a LL compiler (based on [knockout-secure-binding](https://github.com/brianmhunt/knockout-secure-binding)), so some edge case parsing issues may crop up
+
```

Comment:
```suggestion
- The `descendantsComplete` Binding doesn't support the `completeOn: 'render'`-Option in with-bindings
```
@brianmhunt Was that done by design? For me okay

Reference:
https://github.com/knockout/knockout/blob/45e86bf0347cf742ca359bab051ea82fdfae7a5d/src/binding/defaultBindings/ifIfnotWith.js#L15

https://knockoutjs.com/documentation/binding-lifecycle-events.html